onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorRecurrentNeuralNetwork.cpp

```diff
@@ -182,21 +182,15 @@ class DmlOperatorRecurrentBase: public DmlOperator, public RecurrentHelper
         }
     }
     
-    bool ActivationNameCompare(const std::string& activationName, const char* attrValue)
+    bool CompareActivationName(std::string_view activationName, std::string_view attrValue)
     {
-        if (activationName.size() != std::char_traits<char>::length(attrValue)) 
+        if (activationName.size() != attrValue.size()) 
```

Comment:
This check is unnecessary because std::equal returns false if the two ranges have different lengths.